apps/bot_manager/lib/bot_state_machine_checker.ex

```diff
@@ -74,24 +81,31 @@ defmodule BotManager.BotStateMachineChecker do
     time_since_last_position_change >= bot_state_machine.time_amount_to_change_position
   end
 
-  defp bot_health_low?(bot_player) do
-    {:player, bot_player_info} = bot_player.aditional_info
-    health_percentage = bot_player_info.health * 100 / bot_player_info.max_health
-
-    health_percentage <= @low_health_percentage
-  end
-
   defp bot_can_turn_aggresive?(bot_state_machine) do
     bot_state_machine.progress_for_basic_skill >= bot_state_machine.cap_for_basic_skill ||
       bot_state_machine.progress_for_ultimate_skill >= bot_state_machine.cap_for_ultimate_skill
   end
 
   defp bot_can_follow_a_player?(bot_player, bot_state_machine, players) do
     players_nearby_to_follow =
-      Utils.map_directions_to_players(players, bot_player, bot_state_machine.max_vision_range_to_follow_player)
+      Utils.map_directions_to_players(
+        players,
+        bot_player,
+        if(bot_state_machine.is_melee,
+          do: bot_state_machine.melee_tracking_range,
+          else: bot_state_machine.ranged_tracking_range
```

Comment:
Same for this one to check if "is_melee"

Reply:
sure!